lib/src/facts/linux/operating_system_resolver.cc

```diff
@@ -224,9 +224,11 @@ namespace facter { namespace facts { namespace linux {
 
         // For VMware ESX, execute the vmware tool
         if (value.empty() && name == os::vmware_esx) {
-            auto result = execute("vmware", { "-v" });
-            if (result.first) {
-                re_search(result.second, boost::regex("VMware ESX .*?(\\d.*)"), &value);
+            bool success;
+            string output, error;
+            tie(success, output, error) = execute("vmware", { "-v" });
```

Comment:
Error redirects to null, so I'd rather have a variable name that reflects that choice here.